packages/dds/sequence/src/test/intervalCollection.undoRedo.spec.ts

```diff
@@ -0,0 +1,119 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import {
+	MockContainerRuntimeFactory,
+	MockFluidDataStoreRuntime,
+	MockStorage,
+} from "@fluidframework/test-runtime-utils";
+import {
+	appendToMergeTreeDeltaRevertibles,
+	MergeTreeDeltaRevertible,
+	revertMergeTreeDeltaRevertibles,
+} from "@fluidframework/merge-tree";
+import { SharedString } from "../sharedString";
+import { IntervalCollection, IntervalType, SequenceInterval } from "../intervalCollection";
+import { SharedStringFactory } from "../sequenceFactory";
+import { assertIntervals } from "./intervalUtils";
+
+describe("Undo/redo for interval collection operations", () => {
+	let sharedString: SharedString;
+	let dataStoreRuntime1: MockFluidDataStoreRuntime;
+	let containerRuntimeFactory: MockContainerRuntimeFactory;
+	let collection: IntervalCollection<SequenceInterval>;
+	let revertibles: MergeTreeDeltaRevertible[];
+
+	beforeEach(() => {
+		dataStoreRuntime1 = new MockFluidDataStoreRuntime({ clientId: "1" });
+		sharedString = new SharedString(
+			dataStoreRuntime1,
+			"shared-string-1",
+			SharedStringFactory.Attributes,
+		);
+
+		containerRuntimeFactory = new MockContainerRuntimeFactory();
+
+		// Connect the first SharedString.
+		dataStoreRuntime1.local = false;
+		const containerRuntime1 = containerRuntimeFactory.createContainerRuntime(dataStoreRuntime1);
+		const services1 = {
+			deltaConnection: containerRuntime1.createDeltaConnection(),
+			objectStorage: new MockStorage(),
+		};
+		sharedString.initializeLocal();
+		sharedString.connect(services1);
+		collection = sharedString.getIntervalCollection("test");
+		revertibles = [];
+	});
+
+	it("has an interval contained within the deleted range", () => {
+		sharedString.insertText(0, "hello world");
+
+		sharedString.on("sequenceDelta", (op) => {
+			appendToMergeTreeDeltaRevertibles(sharedString, op.deltaArgs, revertibles);
+		});
+
+		collection.add(2, 4, IntervalType.SlideOnRemove);
+
+		sharedString.removeRange(0, 6);
+
+		revertMergeTreeDeltaRevertibles(sharedString, revertibles.splice(0));
+
+		assert.equal(sharedString.getText(), "hello world");
+		assertIntervals(sharedString, collection, [{ start: 6, end: 6 /* start: 2, end: 4 */ }]);
+	});
+	it("has an interval with both endpoints contained within the deleted range", () => {
```

Comment:
I'm not sure what this test is meant to cover different from the previous one. That the interval start is the same index as the start of the removed segment?

Reply:
It's meant to have an interval with the same range as the deleted text, but I forgot to update the endpoint based on Abram's notes from the other thread. I've changed the test now so the interval endpoint is the same index as the end of the deleted range.